releasenotes-builder/src/test/java/com/github/checkstyle/MainTest.java

```diff
@@ -0,0 +1,217 @@
+                .thenReturn(mockGhRepository);
+        // CsGit static
+        final MockedStatic<CsGit> mockCsGitStatic = mockStatic(CsGit.class);
+        mockCsGitStatic.when(
+                () -> CsGit.getCommitsBetweenReferences(anyString(), anyString(), anyString()))
+                .thenReturn(TEST_COMMITS);
+
+        // TwitterPublisher static
+        mockStatic(TwitterPublisher.class);
+    }
+
+    @Before
+    public void reset() {
+        TEST_COMMITS.clear();
+    }
+
+    @Test
+    public void testNoCommits() throws Exception {
+        assertMainReturnCode(0,
+                "-localRepoPath", temporaryFolder.newFolder().getAbsolutePath(),
+                "-remoteRepoPath", "checkstyle/checkstyle",
+                "-startRef", "12345678",
+                "-releaseNumber", "10.0.1",
+                "-outputLocation", temporaryFolder.newFolder().getAbsolutePath(),
+                "-githubAuthToken", "TOKEN",
+                "-generateAll",
+                "-publishTwit",
+                "-twitterConsumerKey", "KEY",
+                "-twitterConsumerSecret", "SECRET",
+                "-twitterAccessToken", "TOKEN",
+                "-twitterAccessTokenSecret", "SECRET",
+                "-validateVersion"
+        );
+    }
+
+    @Test
+    public void testUnknownCommit() throws Exception {
+        TEST_COMMITS.add(RevCommitUtil.create("Hello World"));
+
+        assertMainReturnCode(0,
+                "-localRepoPath", temporaryFolder.newFolder().getAbsolutePath(),
+                "-remoteRepoPath", "checkstyle/checkstyle",
+                "-startRef", "12345678",
+                "-releaseNumber", "10.0.1",
+                "-outputLocation", temporaryFolder.newFolder().getAbsolutePath(),
+                "-githubAuthToken", "TOKEN",
+                "-generateAll",
+                "-publishTwit",
+                "-twitterConsumerKey", "KEY",
+                "-twitterConsumerSecret", "SECRET",
+                "-twitterAccessToken", "TOKEN",
+                "-twitterAccessTokenSecret", "SECRET",
+                "-validateVersion"
+        );
+    }
+
+    @Test
+    public void testIssueCommitWithIssueNotFound() throws IOException {
+        TEST_COMMITS.add(RevCommitUtil.create("Issue #1: Hello World"));
+
+        assertMainReturnCode(-2,
+                "-localRepoPath", temporaryFolder.newFolder().getAbsolutePath(),
+                "-remoteRepoPath", "checkstyle/checkstyle",
+                "-startRef", "12345678",
+                "-releaseNumber", "10.0.1",
+                "-outputLocation", temporaryFolder.newFolder().getAbsolutePath(),
+                "-githubAuthToken", "TOKEN",
+                "-generateAll",
+                "-publishTwit",
+                "-twitterConsumerKey", "KEY",
+                "-twitterConsumerSecret", "SECRET",
+                "-twitterAccessToken", "TOKEN",
+                "-twitterAccessTokenSecret", "SECRET",
+                "-validateVersion"

```

Comment:
Please make one more assertMainReturnCode
That set only version and validateVersion, all other parameters are noice in such test cases

Reply:
I believe this is done. Let me know if I am not understanding this.
````
    @Test
    public void testValidateVersion() {
        runMainAndAssertReturnCode(0,
                "-releaseNumber", "10.0.1",
                "-validateVersion"
        );
    }
````